### PR TITLE
monaco can react to work space language changes

### PR DESCRIPTION
**Pull request by colombod**
when loading workspaces with language feature monaco changes grammar

### CODE REVIEW

File: Microsoft.DotNet.Try.Client/src/components/Editor.tsx [hunk 1]
```diff
@@ -188,6 +156,51 @@ export class Editor extends React.Component<IEditorProps, IEditorState> {
             .subscribe((event) => this.textChangedEventHandler(event));
     }
 
+    private setupLangaugeServices = (monaco: typeof monacoEditor, language: SupportedLanguages, capturedEditor: Editor) => {
```
Comment (jonsequitur) [inline Microsoft.DotNet.Try.Client/src/components/Editor.tsx]: Spelling.

Comment (colombod) [reply Microsoft.DotNet.Try.Client/src/components/Editor.tsx]: and i checked all over for that one ...

File: Microsoft.DotNet.Try.Client/test/ActionCreators/workspaceActionCreators.specs.ts [hunk 2]
```diff
@@ -27,50 +27,55 @@ suite("Workspace Action Creators", () => {
...
                 files: [],
                 buffers: [{ id: "Program.cs", content: "", position: 0 }],
                 usings: [],
+                language: "csharp"
             }),
             actions.setActiveBuffer("Program.cs"),
             actions.setCodeSource("http://source.com")
         ];
-        configureWorkspace(store, undefined, undefined, encodeURIComponent("http://source.com"));
+        configureWorkspace({ store, fromParameter: encodeURIComponent("http://source.com") });
```
Comment (jonsequitur) [inline Microsoft.DotNet.Try.Client/test/ActionCreators/workspaceActionCreators.specs.ts]: Better to use a domain we own and which helps illustrate the use case better.

File: Microsoft.DotNet.Try.Client/src/actionCreators/workspaceActionCreators.ts [hunk 3]
```diff
@@ -104,59 +114,65 @@ export const LoadWorkspaceFromGist: ActionCreator<ThunkAction<Promise<Action>, I
...
-            workspace.workspaceType = decodeURIComponent(workspaceTypeParameter);
+        if (configuration.workspaceTypeParameter) {
+            workspace.workspaceType = decodeURIComponent(configuration.workspaceTypeParameter);
         }
     }
 
-    store.dispatch(setWorkspaceType(workspace.workspaceType));
-    store.dispatch(setWorkspace(workspace));
-    store.dispatch(setActiveBuffer(bufferId));
+    if (!isNullOrUndefinedOrWhitespace(configuration.language)) {
```
[the remark this answers is not in the document]
Comment (colombod) [reply Microsoft.DotNet.Try.Client/src/actionCreators/workspaceActionCreators.ts]: whitespace evaluates as true

**Pull request merged**